### ISSUE TITLE
Null terminators showing up randomly in JSON results for numeric fields - db2util 1.0.11-dev

### ISSUE THREAD

**Issue by richardschoen** [IBM/ibmi-db2util#14]
Attached JSON output with random null terminators on last two fields.
[qcustcdt.zip](https://github.com/IBM/ibmi-db2util/files/6802865/qcustcdt.zip)

Used sample file QIWS/QCUSTCDT

**Sample Command Line**
db2util -o json "select * from qiws.qcustcdt"  > /tmp/qcustcdt.json

utility version: db2util 1.0.11-dev

**Comment by richardschoen** [comment]
Checked without JSON and got same error in results data.
db2util  "select * from qiws.qcustcdt"  > /tmp/qcustcdt.txt

**Comment by kadler** [comment]
This was introduced by #12. We now adjust the length in db2util.c based on col->data, but we change col->data if it's a numeric type based on the value returned in the bound buffer. Prevously, we only did the strlen on the finalized calculated buffer passed to the format functions and these values are always based on the underlying buffer pointer, which is never adjusted.

We either need to save off the actual pointer we passed to SQLBindCol and pass that to strlen() or re-adjust col->data back. I think the code is tricky enough doing these adjustments, so saving off makes the most sense to me.
